var/spack/repos/builtin/packages/py-frozenlist/package.py

```diff
@@ -15,14 +15,22 @@ class PyFrozenlist(PythonPackage):
 
     license("Apache-2.0")
 
+    version("1.4.1", sha256="c037a86e8513059a2613aaba4d817bb90b9d9b6b69aace3ce9c877e8c8ed402b")
     version("1.3.1", sha256="3a735e4211a04ccfa3f4833547acdf5d2f863bfeb01cfd3edaffbc251f15cec8")
     version("1.3.0", sha256="ce6f2ba0edb7b0c1d8976565298ad2deba6f8064d2bebb6ffce2ca896eb35b0b")
     version("1.2.0", sha256="68201be60ac56aff972dc18085800b6ee07973c49103a8aba669dee3d71079de")
 
     depends_on("c", type="build")  # generated
 
+    # 1.4.0 is the first version to support python 3.12
+    # https://github.com/aio-libs/frozenlist/issues/433#issuecomment-1633197557
+    conflicts("^python@3.12:", when="@:1.4.0")
```

Comment:
```suggestion
    conflicts("^python@3.12:", when="@:1.3")
```
Version ranges are inclusive